src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxePsiFieldImpl.java

```diff
@@ -62,25 +64,16 @@ public String getName() {
   }
 
   @Override
-  @NotNull
+  @Nullable
   public HaxeComponentName getComponentName() {
-    return new HaxeComponentNameImpl(getNode());
-    ////this.getName();
-    //return findNotNullChildByClass(HaxeComponentName.class);
+    return null;
```

Comment:
> Will cause endless recursion. 

Using your new code, yes.  I was browsing my sources, not the change list, so I got confused.  However, now that the getNameIdentifier() calls getComponentName(), and getComponentName() always returns null, we've stripped the implementation out of *two* functions.

> Notice that getNameIdentifier is declared for Mixin

I don't see a mixin class anywhere in the superclass hierarchy.  Are you saying that *all* subclasses implement it?  If we are going to depend upon that, then we should just remove the method from this (abstract) class; then we'll enforce it.